core/src/main/scala/org/apache/spark/deploy/history/EventLogFileCompactor.scala

```diff
@@ -0,0 +1,239 @@
+    results.isEmpty || results.forall(_ == true)
+  }
+
+  private def applyFilter(filter: EventFilter, event: SparkListenerEvent): Option[Boolean] = {
+    event match {
+      case event: SparkListenerStageSubmitted => filter.filterStageSubmitted(event)
+      case event: SparkListenerStageCompleted => filter.filterStageCompleted(event)
+      case event: SparkListenerJobStart => filter.filterJobStart(event)
+      case event: SparkListenerJobEnd => filter.filterJobEnd(event)
+      case event: SparkListenerTaskStart => filter.filterTaskStart(event)
+      case event: SparkListenerTaskGettingResult => filter.filterTaskGettingResult(event)
+      case event: SparkListenerTaskEnd => filter.filterTaskEnd(event)
+      case event: SparkListenerEnvironmentUpdate => filter.filterEnvironmentUpdate(event)
+      case event: SparkListenerBlockManagerAdded => filter.filterBlockManagerAdded(event)
+      case event: SparkListenerBlockManagerRemoved => filter.filterBlockManagerRemoved(event)
+      case event: SparkListenerUnpersistRDD => filter.filterUnpersistRDD(event)
+      case event: SparkListenerApplicationStart => filter.filterApplicationStart(event)
+      case event: SparkListenerApplicationEnd => filter.filterApplicationEnd(event)
+      case event: SparkListenerExecutorMetricsUpdate => filter.filterExecutorMetricsUpdate(event)
+      case event: SparkListenerStageExecutorMetrics => filter.filterStageExecutorMetrics(event)
+      case event: SparkListenerExecutorAdded => filter.filterExecutorAdded(event)
+      case event: SparkListenerExecutorRemoved => filter.filterExecutorRemoved(event)
+      case event: SparkListenerExecutorBlacklistedForStage =>
+        filter.filterExecutorBlacklistedForStage(event)
+      case event: SparkListenerNodeBlacklistedForStage =>
+        filter.filterNodeBlacklistedForStage(event)
+      case event: SparkListenerExecutorBlacklisted => filter.filterExecutorBlacklisted(event)
+      case event: SparkListenerExecutorUnblacklisted => filter.filterExecutorUnblacklisted(event)
+      case event: SparkListenerNodeBlacklisted => filter.filterNodeBlacklisted(event)
+      case event: SparkListenerNodeUnblacklisted => filter.filterNodeUnblacklisted(event)
+      case event: SparkListenerBlockUpdated => filter.filterBlockUpdated(event)
+      case event: SparkListenerSpeculativeTaskSubmitted =>
+        filter.filterSpeculativeTaskSubmitted(event)
+      case _ => filter.filterOtherEvent(event)
+    }
+  }
+}
+
+class CompactedEventLogFileWriter(
+    originalFilePath: Path,
+    appId: String,
+    appAttemptId: Option[String],
+    logBaseDir: URI,
+    sparkConf: SparkConf,
+    hadoopConf: Configuration)
+  extends SingleEventLogFileWriter(appId, appAttemptId, logBaseDir, sparkConf, hadoopConf) {
+
+  override val logPath: String = originalFilePath.toUri.toString + EventLogFileWriter.COMPACTED
+
+  override def writeLine(line: String, flushLogger: Boolean): Unit = {
+    super.writeLine(line, flushLogger)
+  }
```

Comment:
Ah that seems to be missed after refactoring. Will remove.

Reply:
My bad, in base class it's defined as `protected` which is correct as we don't want to allow event writer to write a line instead of json (though we know it's just a line). This class is a variant of writer which should write a line "as it is", instead of rebuilding json and writing it - hence I decide to not touch a base class and only expose the method for this class.

Reply:
So CompactedEventLogFileWriter is just one of exceptional case to not repeat the code in EventLogFileWriter. I didn't want to let CompactedEventLogFileWriter extend EventLogFileWriter but then lots of code will be redundant so had to have one exception.

By intention EventLogFileWriter should still hide the method. If we are not feeling convenient with this, I'll change CompactedEventLogFileWriter to not extend EventLogFileWriter, but try to share the code.

Reply:
I'm changing my mind to just think every line is json which should be mostly correct - will call `writeEvent` instead.